application/src/main/java/org/opentripplanner/model/plan/WalkStep.java

```diff
@@ -130,6 +134,13 @@ public String getExit() {
     return exit;
   }
 
+  /**
+   * Entity related to a step e.g. building entrance/exit.
+   */
+  public Object getEntity() {
+    return entity;
+  }
+
```

Comment:
I would like to se a diagram with boxes - forget about the class design for a moment and just draw the things we want to add in the future (next 1- 2 years). Then we can discuss how to group them and map the relationship. 

Return an Object is on-go, so is `Entrance extends StepEntity`.

Reply:
I drew some sort of a process diagram with boxes together with @HenrikSundell.

![entity-diagram](https://github.com/user-attachments/assets/9a1b6c1a-ddbf-4cda-83de-4b1f66f1c2b9)

Reply:
I just quickly tested and seems like it's possible to define a class that is used for an union from GraphQL (through codegen configuration). So with this current setup where the different entity types would have a shared base class, you could use that for the union. However, if they don't, then you need to return an Object from the `stepImpl#entity()` method.

Reply:
New diagram 
![step-elements](https://github.com/user-attachments/assets/90884319-62c3-4ad7-9ca6-05fd6c5fe4fb)